src/Installer/core-sdk-tasks/GenerateRuntimeAnalyzersSWR.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class GenerateRuntimeAnalyzersSWR : Task
+    {
+        [Required]
+        public string RuntimeAnalyzersLayoutDirectory { get; set; }
+
+        [Required]
+        public string OutputFile { get; set; }
+
+        public override bool Execute()
+        {
+            StringBuilder sb = new StringBuilder(SWR_HEADER);
+
+            // NOTE: Keep in sync with SdkAnalyzerAssemblyRedirector.
+            // This is intentionally short to avoid long path problems.
+            const string installDir = @"DotNetRuntimeAnalyzers";
+
+            AddFolder(sb,
+                      @"AnalyzerRedirecting",
+                      @"Common7\IDE\CommonExtensions\Microsoft\AnalyzerRedirecting",
+                      filesToInclude:
+                      [
+                          "Microsoft.Net.Sdk.AnalyzerRedirecting.dll",
+                          "Microsoft.Net.Sdk.AnalyzerRedirecting.pkgdef",
+                          "extension.vsixmanifest",
+                      ]);
+
+            AddFolder(sb,
+                      @"AspNetCoreAnalyzers",
+                      @$"{installDir}\AspNetCoreAnalyzers");
+
+            AddFolder(sb,
+                      @"NetCoreAnalyzers",
+                      @$"{installDir}\NetCoreAnalyzers");
+
+            AddFolder(sb,
+                      @"WindowsDesktopAnalyzers",
+                      @$"{installDir}\WindowsDesktopAnalyzers");
+
+            AddFolder(sb,
+                      @"SDKAnalyzers",
+                      @$"{installDir}\SDKAnalyzers");
+
+            AddFolder(sb,
+                      @"WebSDKAnalyzers",
+                      @$"{installDir}\WebSDKAnalyzers");
+
+            File.WriteAllText(OutputFile, sb.ToString());
+
+            return true;
+        }
+
+        private void AddFolder(StringBuilder sb, string relativeSourcePath, string swrInstallDir, bool ngenAssemblies = true, IEnumerable<string> filesToInclude = null)
+        {
+            string sourceFolder = Path.Combine(RuntimeAnalyzersLayoutDirectory, relativeSourcePath);
+
+            // If files were specified explicitly, check that they exist.
+            if (filesToInclude != null)
+            {
+                foreach (var file in filesToInclude)
+                {
+                    var path = Path.Combine(sourceFolder, file);
+                    if (!File.Exists(path))
+                    {
+                        throw new InvalidOperationException($"File not found: {path}");
```

Comment:
Do we usually add exception strings as resource strings in this repo? Is it ok to have it hardcoded in English?

Reply:
This is a build-only code, it's not shipped as part of the product, so I would say it's fine.